packages/amplify-category-api/src/graphql-transformer/transformer-options-v2.ts

```diff
@@ -0,0 +1,300 @@
+  const sanityCheckRules = getSanityCheckRules(isNewAppSyncAPI, ff, allowDestructiveUpdates);
+  let resolverConfig = {};
+  if (!_.isEmpty(resources)) {
+    resolverConfig = await context.amplify.invokePluginMethod(
+      context,
+      AmplifyCategories.API,
+      AmplifySupportedService.APPSYNC,
+      'getResolverConfig',
+      [context, resources[0].resourceName],
+    );
+  }
+
+  /**
+   * if Auth is not migrated , we need to fetch resolver Config from transformer.conf.json
+   * since above function will return empty object
+   */
+  if (_.isEmpty(resolverConfig)) {
+    resolverConfig = project.config.ResolverConfig;
+  }
+
+  const buildConfig: TransformerProjectOptions<TransformerFactoryArgs> = {
+    ...options,
+    buildParameters,
+    projectDirectory: resourceDir,
+    transformersFactory: transformerListFactory,
+    transformersFactoryArgs: {
+      addSearchableTransformer: searchableTransformerFlag,
+      storageConfig,
+      authConfig,
+      adminRoles,
+      identityPoolId,
+    },
+    rootStackFileName: 'cloudformation-template.json',
+    currentCloudBackendDirectory: previouslyDeployedBackendDir,
+    minify: options.minify,
+    projectConfig: project,
+    lastDeployedProjectConfig,
+    authConfig,
+    sandboxModeEnabled,
+    sanityCheckRules,
+    resolverConfig,
+  };
+
+  return buildConfig;
+};
+
+const getBucketName = (s3ResourceName: string): { bucketName: string } => {
+  const amplifyMeta = stateManager.getMeta();
+  const stackName = amplifyMeta.providers.awscloudformation.StackName;
+  const s3ResourcePath = pathManager.getResourceDirectoryPath(undefined, AmplifyCategories.STORAGE, s3ResourceName);
+  const cliInputsPath = path.join(s3ResourcePath, 'cli-inputs.json');
+  let bucketParameters: $TSObject;
+  // get bucketParameters 1st from cli-inputs , if not present, then parameters.json
+  if (fs.existsSync(cliInputsPath)) {
+    bucketParameters = JSONUtilities.readJson(cliInputsPath);
+  } else {
+    bucketParameters = stateManager.getResourceParametersJson(undefined, AmplifyCategories.STORAGE, s3ResourceName);
+  }
+  const bucketName = stackName.startsWith('amplify-')
+    ? `${bucketParameters.bucketName}\${hash}-\${env}`
+    : `${bucketParameters.bucketName}${s3ResourceName}-\${env}`;
+  return { bucketName };
+};
+
+const getPreviousDeploymentRootKey = async (previouslyDeployedBackendDir: string): Promise<string|undefined> => {
+  // this is the function
+  let parameters;
+  try {
+    const parametersPath = path.join(previouslyDeployedBackendDir, `build/${PARAMETERS_FILENAME}`);
+    const parametersExists = fs.existsSync(parametersPath);
+    if (parametersExists) {
+      const parametersString = await fs.readFile(parametersPath);
+      parameters = JSON.parse(parametersString.toString());
+    }
+    return parameters.S3DeploymentRootKey;
+  } catch (err) {
+    return undefined;
```

Comment:
Should we handle the file access errors? This seems to be okay to return `undefined` if the file doesn't exist but how about (1) file is locked by some other process or (2) file is not a valid JSON.

Reply:
Possibly, I'm not sure exactly what we'd want to do differently in terms of throwing different errors/error handling, since it would be a DX change. This chunk of code isn't something I wrote, I only moved it, so it won't change existing behavior